Assets/Runtime/core/GameManager.cs

```diff
@@ -1,18 +1,11 @@
 ﻿using System;
 using UnityEngine;
 
-public class GameManager : MonoBehaviour {
-  private static GameManager manager = null;
+public class GameManager : SingletonMonoBehavior<GameManager> {
   private GameState state = GameState.ACTIVE;
 
   private void Awake() {
-    if (manager == null) {
-      GameManager.Manager = this;
-      DontDestroyOnLoad(manager);
-      this.Initialize();
-    } else if (manager != this) {
-      Destroy(this.gameObject); // Delete current instance to maintain singleton.
-    }
+    this.Initialize();
```

Comment:
The null check safety around Initialize is necessary here. GameManager's Initialize should only ever be called once no matter what. Without a safety check, in the off chance that there are multiple GameManagers in the scene there will be unintended effects (such as double bindings being registered). You can quickly test this by adding two game managers on the scene on master branch vs this branch.

Reply:
Good catch - will fix.

Reply:
Fixed.